### ISSUE TITLE
[BUG]Discontinuity learning rate while resume from checkpoint

### ISSUE THREAD

**Issue by zyh3826** [microsoft/DeepSpeed#5788]
**Describe the bug**
My training was broken for some reason. The model is `Llama2-7b base`.
I used `model.save_checkpoint(output_dir)` to save checkpoint, in which `model` is an `DeepSpeedEngine` instance. And my loading function like this:
```python
def deepspeed_load_checkpoint(deepspeed_engine: DeepSpeedEngine, checkpoint_path: str):
    # it's possible that the user is trying to resume from model_path, which doesn't necessarily
    # contain a deepspeed checkpoint. e.g. examples just check if the dir exists and assume it's
    # a resume from a checkpoint and not just a local pretrained weight. So we check here if the
    # path contains what looks like a deepspeed checkpoint
    import glob

    deepspeed_checkpoint_dirs = sorted(glob.glob(f'{checkpoint_path}/global_step*'))

    if len(deepspeed_checkpoint_dirs) > 0:
        print_rank_0(f'Attempting to resume from {checkpoint_path}')
        # this magically updates self.optimizer and self.lr_scheduler
        load_path, _ = deepspeed_engine.load_checkpoint(
            checkpoint_path, load_optimizer_states=True, load_lr_scheduler_states=True
        )
        if load_path is None:
            raise ValueError(f'[deepspeed] failed to resume from checkpoint {checkpoint_path}')
    else:
        raise ValueError(f"Can't find a valid checkpoint at {checkpoint_path}")
```
when I was training for some time, I found that in tensorboard, the learning rate loaded from checkpoint is different from the one before broken, see the figure below, the pink curve means before broken, and the orange curve means after broken loading from checkpoint.
![image](https://github.com/user-attachments/assets/dd01c248-810f-4d3f-8d1c-95cc8613e207)

I wanted to know why this happened. 
Saving the checkpoint I use DeepSpeed version 0.14.2, and loading the checkpoint I use version 0.14.4, does this matter?

**ds_report output**
Please run `ds_report` to give us details about your setup.
```txt
[2024-07-22 17:13:32,781] [INFO] [real_accelerator.py:203:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[42] [93m [WARNING] [0m async_io requires the dev libaio .so object and headers but these were not found.
[43] [93m [WARNING] [0m async_io: please install the libaio-dev package with apt
[44] [93m [WARNING] [0m If libaio is already installed (perhaps from source), try setting the CFLAGS and LDFLAGS environment variables to where it can be found.
[45] [93m [WARNING] [0m Please specify the CUTLASS repo directory as environment variable $CUTLASS_PATH
[46] [93m [WARNING] [0m sparse_attn requires a torch version >= 1.5 and < 2.0 but detected 2.1
[47] [93m [WARNING] [0m using untested triton version (2.1.0), only 1.0.0 is known to be compatible
[48] --------------------------------------------------
[49] DeepSpeed C++/CUDA extension op report
[50] --------------------------------------------------
[51] NOTE: Ops not installed will be just-in-time (JIT) compiled at
[52]       runtime if needed. Op compatibility means that your system
[53]       meet the required dependencies to JIT install the op.
[54] --------------------------------------------------
[55] JIT compiled ops requires ninja
[56] ninja .................. [92m[OKAY][0m
[57] --------------------------------------------------
[58] op name ................ installed .. compatible
[59] --------------------------------------------------
[60] [93m [WARNING] [0m async_io requires the dev libaio .so object and headers but these were not found.
[61] [93m [WARNING] [0m async_io: please install the libaio-dev package with apt
[62] [93m [WARNING] [0m If libaio is already installed (perhaps from source), try setting the CFLAGS and LDFLAGS environment variables to where it can be found.
[63] async_io ............... [93m[NO][0m ....... [93m[NO][0m
[64] fused_adam ............. [93m[NO][0m ....... [92m[OKAY][0m
[65] cpu_adam ............... [93m[NO][0m ....... [92m[OKAY][0m
[66] cpu_adagrad ............ [93m[NO][0m ....... [92m[OKAY][0m
[67] cpu_lion ............... [93m[NO][0m ....... [92m[OKAY][0m
[68] [93m [WARNING] [0m Please specify the CUTLASS repo directory as environment variable $CUTLASS_PATH
[69] evoformer_attn ......... [93m[NO][0m ....... [93m[NO][0m
[70] fp_quantizer ........... [93m[NO][0m ....... [92m[OKAY][0m
[71] fused_lamb ............. [93m[NO][0m ....... [92m[OKAY][0m
[72] fused_lion ............. [93m[NO][0m ....... [92m[OKAY][0m
[73] inference_core_ops ..... [93m[NO][0m ....... [92m[OKAY][0m
[74] cutlass_ops ............ [93m[NO][0m ....... [92m[OKAY][0m
[75] transformer_inference .. [93m[NO][0m ....... [92m[OKAY][0m
[76] quantizer .............. [93m[NO][0m ....... [92m[OKAY][0m
[77] ragged_device_ops ...... [93m[NO][0m ....... [92m[OKAY][0m
[78] ragged_ops ............. [93m[NO][0m ....... [92m[OKAY][0m
[79] random_ltd ............. [93m[NO][0m ....... [92m[OKAY][0m
[80] [93m [WARNING] [0m sparse_attn requires a torch version >= 1.5 and < 2.0 but detected 2.1
[81] [93m [WARNING] [0m using untested triton version (2.1.0), only 1.0.0 is known to be compatible
[82] sparse_attn ............ [93m[NO][0m ....... [93m[NO][0m
[83] spatial_inference ...... [93m[NO][0m ....... [92m[OKAY][0m
[84] transformer ............ [93m[NO][0m ....... [92m[OKAY][0m
[85] stochastic_transformer . [93m[NO][0m ....... [92m[OKAY][0m
[86] --------------------------------------------------
[87] DeepSpeed general environment info:
[88] torch install path ............... ['/opt/conda/lib/python3.10/site-packages/torch']
[89] torch version .................... 2.1.0
[90] deepspeed install path ........... ['/opt/conda/lib/python3.10/site-packages/deepspeed']
[91] deepspeed info ................... 0.14.2, unknown, unknown
[92] torch cuda version ............... 12.1
[93] torch hip version ................ None
[94] nvcc version ..................... 12.1
[95] deepspeed wheel compiled w. ...... torch 2.1, cuda 12.1
[96] shared memory (/dev/shm) size .... 125.78 GB
```

**System info (please complete the following information):**
 - OS: Ubuntu 18.04
 - GPU count and types: one machine with x8 A800s
 - Python version: Python 3.10.13

**Launcher context**
Are you launching your experiment with the `deepspeed` launcher, MPI, or something else?
using `deepspeed` launcher

**Comment by zyh3826** [comment]
Using the same DeepSpeed version 0.14.2 also causes the same problem.

**Comment by zyh3826** [comment]
Closed!!!
Something went wrong when initialize `lr scheduler`.
